src/services/crawl-validator/handle_stake_event.service.ts

```diff
@@ -46,79 +45,78 @@ export default class HandleStakeEventService extends BullableService {
     this.logger.info(`startHeight: ${startHeight}, endHeight: ${endHeight}`);
     if (startHeight >= endHeight) return;
 
-    const stakeTxs: any[] = [];
-    const resultTx = await Transaction.query()
-      .joinRelated('events.[attributes]')
-      .select(
-        'transaction.id',
-        'transaction.timestamp',
-        'transaction.height',
-        'events.id as event_id',
-        'events.type',
-        'events:attributes.key',
-        'events:attributes.value',
-        'events:attributes.index'
-      )
-      .whereIn('events.type', this.eventStakes)
-      .andWhere('transaction.height', '>', startHeight)
-      .andWhere('transaction.height', '<=', endHeight)
-      .andWhere('transaction.code', 0);
+    const stakeEvents: any[] = [];
+    const resultEvents = await Event.query()
+      .select('event.id as event_id', 'event.type', 'event.block_height')
+      .withGraphFetched('transaction')
+      .modifyGraph('transaction', (builder) => {
+        builder.select('id', 'timestamp');
+      })
+      .withGraphFetched('attributes')
+      .modifyGraph('attributes', (builder) => {
+        builder.select('key', 'value', 'index').orderBy('index');
+      })
+      .whereIn('event.type', this.eventStakes)
+      .andWhere('event.block_height', '>', startHeight)
+      .andWhere('event.block_height', '<=', endHeight);
 
-    if (resultTx.length > 0) stakeTxs.push(...resultTx);
+    if (resultEvents.length > 0) stakeEvents.push(...resultEvents);
```

Comment:
ủa sao ko query thẳng ra cái stakeEvents?

Reply:
chỗ đó chắc An nó dùng để pass eslint đó :0 em dùng thẳng thấy báo một đống lỗi phía sau 
![image](https://github.com/aura-nw/horoscope-v2/assets/38557844/f6f41fb6-7eb7-4450-81f2-bbcf7e7d9040)

Reply:
undefined thì check thôi cơ mà thế nào gán lại nó lại ko kêu cả nhỉ =))

Reply:
thì chỗ kia nó check length > 0 nên đoạn sau chắc chắn không undefined rồi đó anh, eslint khôn phết

Reply:
thế if length > 0 mới vào đoạn sau là đc
hack não quá

Reply:
thế thì cái if nó to đùng đó anh

Reply:
cái firstValidator nó bảo có thể undefined là đúng đấy, vì dùng find nên có thể ko ra
ko phải do mảng rỗng đâu
còn cái này chắc bị bug eslint nên nó mới cho pass :-?

Reply:
fixed